### PR TITLE
initialize-model

### PR DESCRIPTION
**Pull request by vuongsythanh**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,6 +13,9 @@ gem "uglifier", ">= 1.3.0"
 gem "coffee-rails", "~> 4.2"
 gem "jquery-rails"
 gem "jbuilder", "~> 2.5"
+gem "turbolinks", "~> 5"
```
Comment (nhunght) [inline Gemfile]: bỏ tubolinks

File: app/models/attach_book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class AttachBook < ApplicationRecord
+    belongs_to :book
```
Comment (nhunght) [inline app/models/attach_book.rb]: indent

File: app/models/author.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Author < ApplicationRecord
+  has_many :book, dependent: :destroy
```
Comment (nhunght) [inline app/models/author.rb]: has_many :books, dependent: :destroy

File: app/models/attach_book.rb (hunk 2 above)
Comment (hungnh103) [inline app/models/attach_book.rb]: indent

File: app/models/author.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  has_many :relationship, dependent: :destroy
```
Comment (nhunght) [inline app/models/author.rb]: has_many :relationships, dependent: :destr

File: app/models/book.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class Book < ApplicationRecord
+  has_many :request, dependent: :destroy
```
Comment (nhunght) [inline app/models/book.rb]: has_many :requests, dependent: :destroy

File: app/models/book.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  has_many :review, dependent: :destroy
```
Comment (nhunght) [inline app/models/book.rb]: has_many :reviews, dependent: :destroy

File: app/models/book.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  has_many :comment, dependent: :destroy
```
Comment (nhunght) [inline app/models/book.rb]: has_many :comments, dependent: :destroy

File: app/models/request.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class Request < ApplicationRecord
+    belongs_to :user
```
Comment (hungnh103) [inline app/models/request.rb]: indent

File: app/models/book.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  has_many :like, dependent: :destroy
```
Comment (nhunght) [inline app/models/book.rb]: has_many :likes, dependent: :destroy

File: app/models/book.rb [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  has_many :attack_book, dependent: :destroy
```
Comment (nhunght) [inline app/models/book.rb]: has_many :attack_books, dependent: :destroy

File: app/models/category.rb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+class Category < ApplicationRecord
+  has_many :book, dependent: :destroy
```
Comment (nhunght) [inline app/models/category.rb]: has_many :books, dependent: :destroy

File: app/models/category.rb [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+
+  validates :name, presence: true, length: {maximum: Settings.category.name_size}
+  validates :description, presence: true,
```
Comment (nhunght) [inline app/models/category.rb]: bỏ pressence: true

File: app/models/comment.rb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :book
+
+  validates :content, presence: true,
+    length: {maximum: Settings.comment.content_size}
+  validates :user_id, presence: true
```
Comment (nhunght) [inline app/models/comment.rb]: có validate :user được không nhỉ?

File: app/models/request.rb (hunk 8 above)
Comment (cntt11b1) [inline app/models/request.rb]: indent

File: app/models/comment.rb [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+  validates :book_id, presence: true
```
Comment (nhunght) [inline app/models/comment.rb]: validates :book

File: app/models/publisher.rb [hunk 15]
```diff
@@ -0,0 +1,14 @@
+class Publisher < ApplicationRecord
+  has_many :book, dependent: :destroy
```
Comment (nhunght) [inline app/models/publisher.rb]: has_many :books, dependent: :destroy

File: app/models/publisher.rb [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+  validates :name, presence: true,
+    length: {maximum: Settings.publisher.name_size}
+  validates :address, presence: true,
+    length: {maximum: Settings.publisher.address_size}
+  validates :email, presence: true,
+    length: {maximum: Settings.publisher.email_size}
+  validates :phone, presence: true,
+    length: {maximum: Settings.publisher.phone_size}
+  validates :description, presence: true,
```
Comment (nhunght) [inline app/models/publisher.rb]: bỏ validate cho description

File: app/models/request.rb (hunk 8 above)
Comment (nhunght) [inline app/models/request.rb]: indent

File: app/models/user.rb [hunk 17]
```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  has_many :request, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_many :requests, dependent: :destroy

File: app/models/user.rb [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+  has_may :activity, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_may :activites, dependent: :destroy

File: config/initializers/config.rb [hunk 19]
```diff
@@ -0,0 +1,36 @@
+Config.setup do |config|
+  # Name of the constant exposing loaded settings
+  config.const_name = 'Settings'
```
Comment (hungnh103) [inline config/initializers/config.rb]: double quote

File: app/models/user.rb [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+  has_many :review, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_many :reviews, dependent: :destroy

File: app/models/user.rb [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+  has_many :comment, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_many :comments, dependent: :destroy

File: app/models/user.rb [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+  has_many :like, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_many :likes, dependent: :destroy

File: app/models/user.rb [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+  has_many :relationship, dependent: :destroy
```
Comment (nhunght) [inline app/models/user.rb]: has_many :relationships, dependent: :destroy

File: config/database.yml [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+  <<: *default
+  database: db/development.sqlite3
+
+test:
+  <<: *default
+  database: db/test.sqlite3
+
+production:
+  <<: *default
+  database: db/production.sqlite3
```
Comment (nhunght) [inline config/database.yml]: sao lại chuyển nó về sqlite rồi.
không được đẩy file này lên cơ mà
c rebase rồi fix conflict với develop đi chứ

File: config/initializers/config.rb (hunk 19 above)
Comment (nhunght) [inline config/initializers/config.rb]: dấu nháy đôi

File: test/fixtures/activities.yml [hunk 25]
```diff
@@ -0,0 +1,2 @@
+one: {}
+two: {}
```
Comment (nhunght) [inline test/fixtures/activities.yml]: remove file

File: test/fixtures/attach_books.yml [hunk 26]
```diff
@@ -0,0 +1,11 @@
...
+  publish_date: 2017-02-24 07:06:10
+  number_page: 1
+  limit_date: 1
+  description: MyText
+
+two:
+  publish_date: 2017-02-24 07:06:10
+  number_page: 1
+  limit_date: 1
+  description: MyText
```
Comment (nhunght) [inline test/fixtures/attach_books.yml]: remove file

File: test/fixtures/authors.yml [hunk 27]
```diff
@@ -0,0 +1,7 @@
+one:
+  name: MyString
+  description: MyText
+
+two:
+  name: MyString
+  description: MyText
```
Comment (nhunght) [inline test/fixtures/authors.yml]: remove file

File: test/fixtures/books.yml [hunk 28]
```diff
@@ -0,0 +1,11 @@
...
+  title: MyString
+  description: MyText
+  status: false
+  image: MyString
+
+two:
+  title: MyString
+  description: MyText
+  status: false
+  image: MyString
```
Comment (nhunght) [inline test/fixtures/books.yml]: remove file

File: test/fixtures/categories.yml [hunk 29]
```diff
@@ -0,0 +1,7 @@
+one:
+  name: MyString
+  description: MyText
+
+two:
+  name: MyString
+  description: MyText
```
Comment (nhunght) [inline test/fixtures/categories.yml]: remove file

File: test/fixtures/comments.yml [hunk 30]
```diff
@@ -0,0 +1,5 @@
+one:
+  content: MyText
+
+two:
+  content: MyText
```
Comment (nhunght) [inline test/fixtures/comments.yml]: remove file

File: test/fixtures/likes.yml [hunk 31]
```diff
@@ -0,0 +1,2 @@
+one: {}
+two: {}
```
Comment (nhunght) [inline test/fixtures/likes.yml]: remove file

File: test/fixtures/publishers.yml [hunk 32]
```diff
@@ -0,0 +1,13 @@
...
+  phone: MyString
+  email: MyString
+  description: MyText
+
+two:
+  name: MyString
+  address: MyString
+  phone: MyString
+  email: MyString
+  description: MyText
```
Comment (nhunght) [inline test/fixtures/publishers.yml]: remove file

File: test/fixtures/relationships.yml [hunk 33]
```diff
@@ -0,0 +1,2 @@
+one: {}
+two: {}
```
Comment (nhunght) [inline test/fixtures/relationships.yml]: remove file

File: test/fixtures/requests.yml [hunk 34]
```diff
@@ -0,0 +1,11 @@
...
+  start_day: 2017-02-24 07:07:47
+  end_day: 2017-02-24 07:07:47
+  desciption: MyText
+  status: false
+
+two:
+  start_day: 2017-02-24 07:07:47
+  end_day: 2017-02-24 07:07:47
+  desciption: MyText
+  status: false
```
Comment (nhunght) [inline test/fixtures/requests.yml]: remove file

File: test/fixtures/reviews.yml [hunk 35]
```diff
@@ -0,0 +1,7 @@
+one:
+  rating: 1
+  content: MyText
+
+two:
+  rating: 1
+  content: MyText
```
Comment (nhunght) [inline test/fixtures/reviews.yml]: remove file

File: test/fixtures/users.yml [hunk 36]
```diff
@@ -0,0 +1,19 @@
...
+
+two:
+  name: MyString
+  password_digest: MyString
+  email: MyString
+  address: MyString
+  phone: MyString
+  avatar: MyString
+  is_admin: false
+  remember_pass: false
```
Comment (nhunght) [inline test/fixtures/users.yml]: remove file

File: test/models/activity_test.rb [hunk 37]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class ActivityTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/activity_test.rb]: remove file

File: test/models/attach_book_test.rb [hunk 38]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class AttachBookTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/attach_book_test.rb]: remove file

File: test/models/author_test.rb [hunk 39]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class AuthorTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/author_test.rb]: remove file

File: test/models/book_test.rb [hunk 40]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class BookTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/book_test.rb]: remove file

File: test/models/category_test.rb [hunk 41]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class CategoryTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/category_test.rb]: remove file

File: test/models/comment_test.rb [hunk 42]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class CommentTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/comment_test.rb]: remove file

File: test/models/like_test.rb [hunk 43]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class LikeTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/like_test.rb]: remove file

File: test/models/publisher_test.rb [hunk 44]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class PublisherTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/publisher_test.rb]: remove file

File: test/models/relationship_test.rb [hunk 45]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class RelationshipTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/relationship_test.rb]: remove file

File: test/models/request_test.rb [hunk 46]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class RequestTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/request_test.rb]: remove file

File: test/models/review_test.rb [hunk 47]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class ReviewTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/review_test.rb]: remove file

File: test/models/user_test.rb [hunk 48]
```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class UserTest < ActiveSupport::TestCase
+end
```
Comment (nhunght) [inline test/models/user_test.rb]: remove file

File: config/initializers/config.rb (hunk 19 above)
Comment (cntt11b1) [inline config/initializers/config.rb]: double quotes

**Pull request closed without merging**